src/DeviceDataStreamValues.tsx

```diff
@@ -0,0 +1,219 @@
+              ))}
+              <td>{new Date(tree.timestamp).toLocaleString()}</td>
+            </tr>
+          ))}
+        </tbody>
+      </Table>
+    </>
+  );
+};
+
+interface InterfaceDataProps {
+  interfaceData: AstarteDatastreamIndividualData[] | AstarteDatastreamObjectData[];
+  aggregation: AstarteInterface['aggregation'];
+}
+
+const InterfaceData = ({ interfaceData, aggregation }: InterfaceDataProps): React.ReactElement => {
+  if (interfaceData.length === 0) {
+    return <p>No data in the selected timeframe.</p>;
+  } else if (aggregation === 'individual') {
+    return <IndividualDatastreamTable data={interfaceData as AstarteDatastreamIndividualData[]} />;
+  }
+  return <ObjectDatastreamTable data={interfaceData as AstarteDatastreamObjectData[]} />;
+};
+
+export default (): React.ReactElement => {
+  const { interfaceName = '', deviceId = '', interfaceMajor } = useParams();
+  const astarte = useAstarte();
+  const [requestingData, setRequestingData] = useState(false);
+  const [fetchedData, setFetchedData] = useState<AstarteInterfaceValues | null>(null);
+  const [formAlerts, formAlertsController] = useAlerts();
+  const [interfaceData, setInterfaceData] = useState<AstarteInterface | null>(null);
+
+  const deviceDataFetcher = useFetch(() =>
+    astarte.client.getDeviceData({
+      deviceId,
+      interfaceName,
+    }),
+  );
+
+  const fetchData = async (path?: string, since?: string, to?: string) => {
+    setRequestingData(true);
+    setFetchedData(null);
+    astarte.client
+      .getDeviceData({
+        deviceId,
+        interfaceName,
+        path,
+        since,
+        to,
+      })
+      .then((data) => {
+        setFetchedData(data);
+      })
+      .catch((err) => {
+        formAlertsController.showError(
+          `Could not fetch data to interface: ${err.response.data.errors.detail}`,
+        );
+        setFetchedData([]);
+      })
+      .finally(() => {
+        setRequestingData(false);
+      });
+  };
+
+  useEffect(() => {
+    if (interfaceMajor) {
+      const major = parseInt(interfaceMajor, 10);
+      astarte.client
+        .getInterface({ interfaceName, interfaceMajor: major })
+        .then((data) => setInterfaceData(data));
+    }
+  }, []);
+
+  return (
+    <Container fluid className="p-3">
+      <div className="d-flex justify-content-between">
+        <h2>
+          <BackButton href={`/devices/${deviceId}/interfaces/${interfaceName}/${interfaceMajor}`} />
+          Interface Datastream Data
+        </h2>
+      </div>
+      <AlertsBanner alerts={formAlerts} />
+      <Card className="mt-4">
+        <Card.Header>
+          <span className="font-monospace">{deviceId}</span> /{interfaceName}
+        </Card.Header>
+        <Card.Body>
+          <FilterDatastreamData
+            data={deviceDataFetcher.value}
+            fetchData={fetchData}
+            requestingData={requestingData}
+          />
```

Comment:
With this name, I though the component was meant to filter and return data.
Instead, it renders some UI controls used to configure filter parameters.
I would suggest something like:

```tsx
<FiltersForm
  interfacePaths={interfacePaths}
  onFiltersChange={fetchData}
  isLoading={requestingData}
/>
```

where:
1. we provide just what the component needs, so `interfacePaths` instead of `deviceDataFetcher.value`
2. we name callback props for event handlers following the convention, where the name starts with `on`: `onFiltersChange`